slite.el

```diff
@@ -1,9 +1,58 @@
-;; -*- lexical-binding: t -*-
+;;; slite.el --- Interactively runs your Common Lisp tests  -*- lexical-binding: t -*-
 
-(provide 'slite)
+;; Copyright (C) 2021-2023 Arnold Noronha
+
+;; Author: Arnold Noronha <arnold@tdrhq.com>
+;; Homepage: https://github.com/tdrhq/slite
+;; Keywords: lisp tools
+;; Package-Requires: ((emacs "25.1"))
+;; SPDX-License-Identifier: Apache-2.0
+
+;;; Commentary:
+
+;; Slite stands for SLIme TEst runner. Slite interactively runs
+;; your Common Lisp tests (currently only FiveAM and Parachute are
+;; supported).  It allows you to see the summary of test failures,
+;; jump to test definitions, rerun tests with debugger all from
+;; inside Emacs.
+
+;; You might want to add some key bindings in various Lisp mode maps:
+;;
+;;     (define-key emacs-lisp-mode-map (kbd "C-c v") #'slite-run)
+;;     (define-key lisp-mode-map (kbd "C-c v") #'slite-run)
+;;     (define-key lisp-mode-map (kbd "C-c j")
+;;       #'slite-compile-defun-and-run-tests)
+;;     (with-eval-after-load 'slime
+;;       (define-key slime-mode-map (kbd "C-c v") #'slite-run))
+;;     (with-eval-after-load 'sly
+;;       (define-key sly-mode-map (kbd "C-c v") #'slite-run))
+
+;;; Code:
 
 (require 'cl-lib)
 
+(declare-function sly-compile-defun "sly")
+(declare-function sly-edit-definition "sly")
+(declare-function sly-eval-async "sly")
+(declare-function sly-mode "sly")
+
+(declare-function slime-compile-defun "slime")
+(declare-function slime-edit-definition "slime")
+(declare-function slime-eval-async "slime")
+(declare-function slime-mode "slime")
+
+(defvar slite-results-mode-map
+  (let ((map (make-sparse-keymap)))
+    (set-keymap-parent map 'tabulated-list-mode-map)
```

Comment:
I'm getting an error on this line (Emacs 28.2):

```
Debugger entered--Lisp error: (wrong-type-argument keymapp tabulated-list-mode-map)
  set-keymap-parent((keymap) tabulated-list-mode-map)
  (let ((map (make-sparse-keymap))) (set-keymap-parent map 'tabulated-list-mode-map) (define-key map (kbd "RET") #'slite-describe-result) (define-key map (kbd "<delete>") #'slite-delete-test) (define-key map (kbd "r") #'slite-rerun-in-debugger) (define-key map (kbd "M-.") #'slite-jump-to-test) (define-key map (kbd "g") #'slite-rerun) (define-key map (kbd "C-c v") #'slite-run) map)
  (defvar slite-results-mode-map (let ((map (make-sparse-keymap))) (set-keymap-parent map 'tabulated-list-mode-map) (define-key map (kbd "RET") #'slite-describe-result) (define-key map (kbd "<delete>") #'slite-delete-test) (define-key map (kbd "r") #'slite-rerun-in-debugger) (define-key map (kbd "M-.") #'slite-jump-to-test) (define-key map (kbd "g") #'slite-rerun) (define-key map (kbd "C-c v") #'slite-run) map) "Keymap for `slite-results-mode'.")
  eval-buffer()  ; Reading at buffer position 1992
  funcall-interactively(eval-buffer)
  command-execute(eval-buffer record)
  execute-extended-command(nil "eval-buffer" nil)
  funcall-interactively(execute-extended-command nil "eval-buffer" nil)
  command-execute(execute-extended-command)
```

Reply:
Should this be without the the quote? It seems to work without the quote.

Reply:
Yes. Fixed.